### PR TITLE
Added test for #5091 (Missing Periods for some DateOffsets)

### LINKED ISSUE #5091: Missing Periods for some DateOffsets
There are not corresponding `Period`s for many `DateOffsets`.

For example see how the freq changes:

``` python
ts = pd.date_range('1/1/2012', periods=4, freq=pd.offsets.BQuarterEnd())
per = ts.to_period()
ts_per = per.to_timestamp()

In [210]: ts
Out[210]: 
<class 'pandas.tseries.index.DatetimeIndex'>
[2012-03-30 00:00:00, ..., 2012-12-31 00:00:00]
Length: 4, Freq: BQ-MAR, Timezone: None

In [211]: per
Out[211]: 
<class 'pandas.tseries.period.PeriodIndex'>
freq: Q-DEC
[2012Q1, ..., 2012Q4]
length: 4

In [212]: ts_per
Out[212]: 
<class 'pandas.tseries.index.DatetimeIndex'>
[2012-01-01 00:00:00, ..., 2012-10-01 00:00:00]
Length: 4, Freq: QS-OCT, Timezone: None
```

This issue is alluded to in comment: https://github.com/pydata/pandas/blob/master/pandas/tseries/frequencies.py#L329

### PR DESCRIPTION
**Pull request by gorogoroumaru**
- [x] closes #5091

### CODE REVIEW

File: pandas/tests/series/test_timeseries.py [hunk 1]
```diff
@@ -106,6 +106,13 @@ def test_asfreq_resample_set_correct_freq(self):
         # does .resample() set .freq correctly?
         assert df.resample("D").asfreq().index.freq == "D"
 
+    def test_period_dateoffset_conversion(self):
```
Comment (mroeschke) [inline pandas/tests/series/test_timeseries.py:109]: Could you put this test in `pandas/tests/indexes/period/test_to_timestamp.py`

File: pandas/tests/series/test_timeseries.py [hunk 2]
```diff
@@ -106,6 +106,13 @@ def test_asfreq_resample_set_correct_freq(self):
...
+        ts = pd.date_range('1/1/2012', periods=4, freq=pd.offsets.BQuarterEnd())
```
Comment (mroeschke) [inline pandas/tests/series/test_timeseries.py:110]: Can you add the issue number to this test? `# GH 5091`

File: pandas/tests/series/test_timeseries.py [hunk 3]
```diff
@@ -106,6 +106,13 @@ def test_asfreq_resample_set_correct_freq(self):
...
+        per = ts.to_period()
+        ts_per = per.to_timestamp()
+
+        tm.assert_datetime_array_equal(ts, ts_per)
```
Comment (mroeschke) [inline pandas/tests/series/test_timeseries.py:114]: Could use use `tm.assert_index_equal` and rename the variables so it reads `tm.assert_index_equal(result, expected)`

**Comment by gorogoroumaru** [comment]
I've done all the corrections, but some tests aren't passing.
Maybe, the issue is still remaining.
Here is the error massage.

______________ TestToTimestamp.test_period_dateoffset_conversion _______________
[gw1] darwin -- Python 3.6.10 /Users/runner/miniconda3/envs/pandas-dev/bin/python

self = <pandas.tests.indexes.period.test_to_timestamp.TestToTimestamp object at 0x14edfb390>

    def test_period_dateoffset_conversion(self):
        # GH5091
        expected = date_range("1/1/2012", periods=4, freq=offsets.BQuarterEnd())
        per = expected.to_period()
        result = per.to_timestamp()
    
>       tm.assert_index_equal(result, expected)
E       AssertionError: Index are different
E       
E       Index values are different (100.0 %)
E       [left]:  DatetimeIndex(['2012-01-01', '2012-04-01', '2012-07-01', '2012-10-01'], dtype='datetime64[ns]', freq='QS-OCT')
E       [right]: DatetimeIndex(['2012-03-30', '2012-06-29', '2012-09-28', '2012-12-31'], dtype='datetime64[ns]', freq='BQ-MAR')

**Comment by mroeschke** [comment]
Ah my mistake. When reviewing that issue I too hastily concluded that the issue was solved.

Any interest in helping solving the bug? Otherwise, definitely recommend other issues that are tagged `good first issue`

**Comment by gorogoroumaru** [comment]
Thank you, I will try another issue.

**Pull request closed without merging**